template/Makefile.in

```diff
@@ -388,6 +388,12 @@ prereq: $(srcdir)/$(CONFIGURE)
 incs: id.h
 all-incs: probes.h
 
+generate-abi: PHONY
+	$(Q) $(BASERUBY) $(tooldir)/abi.rb --srcdir="$(srcdir)"
+
+$(hdrdir)/ruby/internal/abi.h: $(tooldir)/abi.rb | generate-abi
```

Comment:
GNU make specific syntax.

Reply:
I'm not sure what you mean. Are you saying order-only prerequisites are a GNU make specific syntax?

Reply:
Yes, AFAIK.

Reply:
I couldn't get this working without using a order-only prerequisite. I'm not very good at writing Makefiles. Do you know of another way?

Reply:
Also, are there systems that don't use GNU make?

Reply:
Yes.  *BSD systems generally use BSD make, and we need to make sure Ruby's makefiles work with both GNU and BSD make.

Reply:
We could remove the pipe and make it a normal prerequisite. The downside is that Ruby will be relinked every time `make` is installed because it would think `abi.h` is out-of-date. I think this is fine. What do you think?

Reply:
I managed to figure out how to use order-only prerequisite only on GNU make. It will fall back to a regular prerequisite on other make (e.g. BSD make).

What I did is I changed `GNUmakefile` to add the following line:

```
ORDER_ONLY_PRERREQ = |
```

Then in the `Makefile`, I do this:

```
internal/abi.h: $(tooldir)/abi.rb $(ORDER_ONLY_PREREQ) generate-abi
```

That way, if `ORDER_ONLY_PRERREQ` is defined, then `generate-abi` will be a order-only prerequisite. Otherwise, it's a normal prerequisite.